### DISCUSSION TITLE
Purpose of having different edge type for reverse edges

### DISCUSSION THREAD

**Discussion by sakunaharinda** [pyg-team/pytorch_geometric#9894] [General]
Hi,

What is the purpose of having a separate edge type with ```_rev_``` like ```('author', 'rev_writes', 'paper')```, which is generated automatically when transforming a heterogeneous graph using ```T.ToUndirected()(data)```, compared to combining reversed edges with edges using something like ```data.edge_index_dict = {key: to_undirected(edge_index) for key, edge_index in data.edge_index_dict.items()}```?

**Comment by Romain-Nicolle** [comment]
Hi, 

I am not an expert but I will try to help.
Basically, in a Heterogeneous Graph, you have as many Adjacency Matrix as you have edge_types, and by default edges are directed. From that there is two possibilities:

## 1. the src and dst nodes have different types

An edge_type is defined by a tuple: node_type of the source_node, _edge_type, node_type of the destination_node. 
In the example that you provide, the basic edge_type is ```('author', 'writes', 'paper')``` which is a directed edge going from ```'author'``` to ```'paper'```. If you want to create the reverse edge_type, you have to create a ```('paper', 'rev_writes', 'author')``` edge because the src and dst node_types are not the same. 

## 2. the src and dst nodes have the same types

For example you have a social network with users and followers. let's say on Instagram for example that you have a edge_type ```('user', 'follows', 'user')```. This is a directed edge_type this is not because I follow you that you follow me back. 
There if you wanted, you could created the reverse connections and concat them to this edge but I am not sure that you want to do that.
Basically in a Heterogeneous Graph, you learn one specific weight matrix for each adjacency matrix during training. And you have one adjacency matrix for each edge_type.

Let's take the example of 'social media influencers', by definition they have millions of followers, but that doesn't mean that they follow them back. There is a direction in the edge, meaning that message passing framework goes from src_node to dst_node. Adding the reverse edges to the same edge_type would change the weights learnt during training.

## conclusion

I don't know what your problem is, maybe it is relevant to use your approach. Maybe try both of them and see what are the changes.
But my expectations are that the results will be lower if you try to add the reverse edges to the same edge_type (only if tthe src and dst nodes have the same node_type), because you will lose discriminative powers between your different nodes. 

Hope that I was clear and that it helps.

best,
